cmake/onnxruntime_providers.cmake

```diff
@@ -176,6 +180,29 @@ if (onnxruntime_USE_TENSORRT)
   endif()
 endif()
 
+if (onnxruntime_USE_NGRAPH)
+  include_directories("${CMAKE_CURRENT_BINARY_DIR}/onnx")
+  file(GLOB_RECURSE onnxruntime_providers_ngraph_cc_srcs
+    "${ONNXRUNTIME_ROOT}/core/providers/ngraph/*.h"
+    "${ONNXRUNTIME_ROOT}/core/providers/ngraph/*.cc"
+  )
+
+  source_group(TREE ${ONNXRUNTIME_ROOT}/core FILES ${onnxruntime_providers_ngraph_cc_srcs})
+  add_library(onnxruntime_providers_ngraph ${onnxruntime_providers_ngraph_cc_srcs})
+  onnxruntime_add_include_to_target(onnxruntime_providers_ngraph onnxruntime_common onnxruntime_framework gsl onnx onnx_proto protobuf::libprotobuf)
+  add_dependencies(onnxruntime_providers_ngraph ngraph onnx ${onnxruntime_EXTERNAL_DEPENDENCIES})
+  set_target_properties(onnxruntime_providers_ngraph PROPERTIES FOLDER "ONNXRuntime")
+  target_include_directories(onnxruntime_providers_ngraph PRIVATE ${ONNXRUNTIME_ROOT} ${ngraph_INCLUDE_DIRS})
+  set_target_properties(onnxruntime_providers_ngraph PROPERTIES LINKER_LANGUAGE CXX)
+
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIE -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security")
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -z noexecstack")
```

Comment:
If you set these flags globally, they will also affect third-party binaries like protoc, which may not be wanted.

Reply:
speaking of protoc, I tried to build this on my dev box and ran into an issue. the protoc that gets built core dumps immediately when run. seems like it's not getting built with pthread. might be related?

Reply:
Through offline debug, we found the issue @jywu-msft encountered is related.